cpp/src/strings/search/findall.cu

```diff
@@ -110,17 +110,18 @@ struct findall_count_fn : public findall_fn<stack_size> {
 
 //
 std::unique_ptr<table> findall(
-  strings_column_view const& strings,
+  strings_column_view const& input,
   std::string const& pattern,
+  regex_flags const flags,
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
 {
-  auto const strings_count = strings.size();
-  auto const d_strings     = column_device_view::create(strings.parent(), stream);
+  auto const strings_count = input.size();
+  auto const d_strings     = column_device_view::create(input.parent(), stream);
 
-  auto const d_flags = detail::get_character_flags_table();
   // compile regex into device object
-  auto const d_prog      = reprog_device::create(pattern, d_flags, strings_count, stream);
+  auto const d_prog =
+    reprog_device::create(pattern, flags, get_character_flags_table(), strings_count, stream);
```

Comment:
Thanks. This is likely a result of code evolution when the code was originally ported. My intention is to look at removing this parameter later on and just have the function grab the table directly since it is just a singleton inside libcudf. This will likely happen as part of the work involved to solve #5217.

Reply:
Sounds good.